Replace deprecated actions/npm in workflow

This action is deprecated in favor of using the run script step and [actions/setup-node](https://github.com/actions/setup-node).

**Additional context**
```
Download action repository 'actions/npm@master'
##[warning]Failed to download action 'https://api.github.com/repos/actions/npm/tarball/master'. Error Response status code does not indicate success: 404 (Not Found).
##[warning]Back off 13.322999999999999 seconds before retry.
##[warning]Failed to download action 'https://api.github.com/repos/actions/npm/tarball/master'. Error Response status code does not indicate success: 404 (Not Found).
##[warning]Back off 29.112 seconds before retry.
##[error]Response status code does not indicate success: 404 (Not Found).
```

--- a/.github/workflows/push.yml
+++ b/.github/workflows/push.yml
@@ -26,11 +26,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@master
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '10.x'
     - name: npm install
-      uses: actions/npm@master
-      with:
-        args: install --no-save
+      run: npm install --no-save
     - name: build frontend
-      uses: actions/npm@master
-      with:
-        args: run production
+      run: npm run production

--- a/.github/workflows/push.yml
+++ b/.github/workflows/push.yml
@@ -6,9 +6,11 @@ jobs:
     steps:
     - uses: actions/checkout@master
     - name: composer install
-      uses: MilesChou/composer-action@master
+      uses: docker://mileschou/composer
       with:
         args: install --prefer-dist --no-ansi --no-interaction --no-progress --no-scripts
+    - name: Fix vendor permissions
+      run: sudo chown -R runner:docker vendor/
     - name: codesniffer
       uses: docker://cytopia/phpcs
       with:

--- a/.github/workflows/push.yml
+++ b/.github/workflows/push.yml
@@ -13,14 +13,13 @@ jobs:
       uses: docker://cytopia/phpcs
       with:
         args: --standard=phpcs.xml
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '10.x'
     - name: twill-install
-      uses: actions/npm@master
-      with:
-        args: run twill-install
+      run: npm run twill-install
     - name: twill-build
-      uses: actions/npm@master
-      with:
-        args: run twill-build
+      run: npm run twill-build
 
   frontend:
     runs-on: ubuntu-latest

